src/core_plugins/console/public/src/__tests__/app.js

```diff
@@ -0,0 +1,84 @@
+import sinon from 'sinon';
+import $ from 'jquery';
+
+import history from '../history';
+import mappings from '../mappings';
+import init from '../app';
+
+describe('app initialization', () => {
+  const sandbox = sinon.sandbox.create();
+
+  let inputMock, outputMock;
+  let ajaxDoneStub;
+  beforeEach(() => {
+    ajaxDoneStub = sinon.stub();
+    sandbox.stub($, 'ajax').returns({ done: ajaxDoneStub });
+    sandbox.stub(history, 'getSavedEditorState');
+    sandbox.stub(mappings, 'startRetrievingAutoCompleteInfo');
+
+    inputMock = {
+      update: sinon.stub(),
+      moveToNextRequestEdge: sinon.stub(),
+      highlightCurrentRequestsAndUpdateActionBar: sinon.stub(),
+      updateActionsBar: sinon.stub(),
+      getSession: sinon.stub().returns({ on() {} })
+    };
+
+    outputMock = {
+      update: sinon.stub()
+    };
+  });
+
+  afterEach(() => {
+    sandbox.restore();
+  });
+
+  it('correctly loads state from any external HTTPS links.', () => {
+    const mockContent = {};
+    ajaxDoneStub.yields(mockContent);
+
+    init(inputMock, outputMock, 'https://state.link.com/content');
+
+    sinon.assert.calledOnce($.ajax);
+    sinon.assert.calledWithExactly($.ajax, {
+      url: 'https://state.link.com/content',
+      dataType: 'text',
+      kbnXsrfToken: false
+    });
+
+    sinon.assert.calledTwice(inputMock.moveToNextRequestEdge);
```

Comment:
__note:__ it's a bit weird that it's called twice (not sure if it's a bug or not), but here is how it works :)